src/Analyzer.Cli.FunctionalTests/CommandLineParserTests.cs

```diff
@@ -116,39 +116,6 @@ public void AnalyzeTemplate_ValidInputValues_AnalyzesUsingAutoDetectedParameters
             StringAssert.Contains(outputWriter.ToString(), "Parameters File: " + Path.Combine(Directory.GetCurrentDirectory(), "Tests", "ToTestSeparateParametersFile", "TemplateWithSeparateParametersFile.parameters.json"));
         }
 
-        [TestMethod]
-        public void AnalyzeDirectory_ValidInputValues_AnalyzesExpectedNumberOfFiles()
-        {
-            var args = new string[] { "analyze-directory", Path.Combine(Directory.GetCurrentDirectory(), "Tests") };
-
-            using StringWriter outputWriter = new();
-            Console.SetOut(outputWriter);
-
-            var result = _commandLineParser.InvokeCommandLineAPIAsync(args);
-
-            Assert.AreEqual((int)ExitCode.ErrorAndViolation, result.Result);
-            StringAssert.Contains(outputWriter.ToString(), "Analyzed 14 files");
```

Comment:
I think we should still validate that that message of the CLI output is correct, and I don't see it in the new tests

Reply:
The way I understood these tests were just to validate that Template Analyzer discovered the right number of templates/parameters, and with the old architecture and test setup, the *only* way to do that was by capturing the CLI output.  Moving this into a unit test (now that it's unit testable) seemed more correct.

For example, "Analyzed 14 files" is not actually the number of templates discovered.  So is it validating that some templates *weren't* analyzed?  That didn't make sense to me.

Reply:
mmm I'm not sure, I just pointed it out because I thought we also wanted to validate all the text that gets printed by the CLI